### DISCUSSION TITLE
[Help Needed] Use text expansions inside rofi

### DISCUSSION THREAD

**Discussion by idr4n** [davatorium/rofi#1315] [Q&A]
I'm just trying to get some help on how to get text expansions to work.

I use a text expansion application called Espanso, which is similar to text expander but cross-platform. Basically, I want to use espanso text expansions inside a `rofi -dmenu` instance.

I have two machines running Arch with `rofi-bin` installed from AUR. In one of them, expansions work without issues inside rofi, that is, if I trigger `rofi -dmenu` and type, e.g., :date,  then the current date is expanded as expected. However, in the machine using the same rofi configuration and same espanso snippets, if I type `:date`, then the current date gets expanded, however Not inside rofi but in the application that was focused when I triggered rofi in the first place, say a terminal instance. If I triger rofi as a normal window (`rofi -dmenu -normal-window`) expansions work without issues.

I really don't know what is causing expansions to work in one machine and not in the other. Again, I'm using the same configuration in both machines to the best of my knowledge but there should be something different that is causing this behavior for sure. I have looked for solutions without any luck. Any ideas on how to get it to work regardless of the machine would be greatly appreciated.

**Comment by DaveDavenport** [comment]
Sorry, missed this topic. 

I can see this not working great,  rofi creates an unmanged window that sits over everything and grabs the keyboard (works slightly differently in -normal-window mode). 
I have no idea how espanso works, but it might not see what is entered in rofi.  I am not surprised it does not work. (also no idea how to get it to work.).

**Reply by idr4n** [reply]
I understand. Thanks for your reply. I still use rofi for mostly everything else so just need to get around entering data directly in the terminal for example whenever I need text expansions.
